src/Modules/UngradedModules/SubQuoHom.jl

```diff
@@ -329,36 +329,98 @@ function is_welldefined(H::SubQuoHom)
   return true
 end
 
-function (==)(f::ModuleFPHom, g::ModuleFPHom)
+# An internal function comparing only the trivial things beyond potential `ring_map`s.
+function _cmp_internal(f::ModuleFPHom, g::ModuleFPHom)
   domain(f) === domain(g) || return false
   codomain(f) === codomain(g) || return false
-  M = domain(f)
-  for v in gens(M)
-    f(v) == g(v) || return false
-  end
+  return all(f(v) == g(v) for v in gens(domain(f)))
+end
+
+# No ring map
+function (==)(f::ModuleFPHom{D, C, Nothing}, g::ModuleFPHom{D, C, Nothing}) where {D, C}
+  return _cmp_internal(f, g)
+end
+
+# With ring map
+# Note that we only allow comparison in case that the ring map is either `nothing` 
+# (i.e. the identity), or an honest `Map`. For these cases, we may require that 
+# comparison `==` is properly implemented. Since the ring maps can take any other 
+# form (e.g. rings by abuse of type-casting, or anonymous julia functions), we have 
+# little chance to write generic code for comparison in such cases. Note that 
+# in many cases, `==` will even default to `===` and then produce mathematically 
+# wrong results (i.e. returning `false` on `f==g` on mathematically equal maps. 
+function (==)(
+              f::ModuleFPHom{D, C, Nothing}, 
+              g::ModuleFPHom{D, C, <:Map}
+             ) where {D, C}
+  is_trivial(ring_map(g)) || return false
+  return _cmp_internal(f, g)
+end
+
+function (==)(
+              f::ModuleFPHom{D, C, <:Map}, 
+              g::ModuleFPHom{D, C, Nothing}
+             ) where {D, C}
+  is_trivial(ring_map(f)) || return false
+  return _cmp_internal(f, g)
+end
+
+function (==)(
+              f::ModuleFPHom{D, C, <:Map}, 
+              g::ModuleFPHom{D, C, <:Map}
+             ) where {D, C}
+  # for `Map`s we expect `==` to be properly implemented
+  ring_map(g) == ring_map(f) || return false
+  return _cmp_internal(f, g)
+end
+
+function (==)(f::ModuleFPHom, g::ModuleFPHom)
+  # `===` is the only generic case we can handle in general; see above comment.
+  is_equal_as_morphism(f, g) || return false # this call might even throw!
   return true
```

Comment:
```suggestion
  return is_equal_as_morphism(f, g) # this call might even throw!
```
that should be equivalent, but more readable